python_modules/dagster/dagster/_core/definitions/asset_check_spec.py

```diff
@@ -31,23 +33,26 @@ class AssetCheckSeverity(Enum):
     ERROR = "ERROR"
 
 
-class AssetCheckSpec(
-    NamedTuple(
-        "_AssetCheckSpec",
-        [
-            ("name", PublicAttr[str]),
-            ("asset_key", PublicAttr[AssetKey]),
-            ("description", PublicAttr[Optional[str]]),
-            ("additional_deps", PublicAttr[Iterable["AssetDep"]]),
-            (
-                "blocking",  # intentionally not public, see https://github.com/dagster-io/dagster/issues/20659
-                bool,
-            ),
-            ("metadata", PublicAttr[Optional[Mapping[str, Any]]]),
-            ("automation_condition", Optional["AutomationCondition[AssetCheckKey]"]),
-        ],
+LazyAutomationCondition: TypeAlias = Annotated[
+    "AutomationCondition",
+    ImportFrom("dagster._core.definitions.declarative_automation.automation_condition"),
+]
+
+LazyAssetDep: TypeAlias = Annotated["AssetDep", ImportFrom("dagster._core.definitions.asset_dep")]
+
+
+@record_custom
```

Comment:
remove all the check calls in the `super().__new__` call since they are duplicative with what `@record` will do